### PR TITLE
Password cracking with john the ripper

### LINKED ISSUE #2567: Password Cracking with John the Ripper
**Introduction**
John the Ripper is a free open source password recovery and cracking security auditing tool. It is available in most operating systems but most recommended in Kali Linux.
John the Ripper is mostly used in penetration testing by security specialist to spot weak passwords in their system. The article proposed will be easily understood from a beginner to an advanced user.
The article will clearly guide the user in a step by step process to chose a target machine and set up necessary options for the bruteforce. The article will help the user avoid major errors caused by users during a brute force attack.

**Disclaimer**
The tutorial is meant for educational purposes only and any action that is related to content in this article is your responsibility.

**Key Takeaways**
By the end of this tutorial, the user will be able to:
1. Install John the Ripper in Kali Linux.
2. Run some major commands used in John the Ripper.
3. Perform a dictionary attack using John the Ripper
4. Understand word mangling rules in JtR.
5. Display their output during an attack.

**Reference**
N/A

### PR DESCRIPTION
**Pull request by Koros-Meshack**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
This fixes #2567

### CODE REVIEW

**Comment by ahmadmardeni1** [comment]
Good afternoon and thank you for submitting your article for publishing.
Your article has been entered into our queue and should be reviewed (and published pending any changes) as soon as a peer reviewer is finished with the ones in the queue before it. :rocket:
Thank you. @Koros-Meshack

**Review by Ericgacoki — CHANGES_REQUESTED** [review]
Greetings @Koros-Meshack, Welcome to Section's Eng-Ed program. We're thrilled to have you onboard.
Please work on the suggestions attached below as we proceed with the review. Feel free to ping me in case of any query.

File: content/articles/password-cracking-with-john-the-ripper/index.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+Now, let's set the password for the account.
+```bash
+$ passwd James
+```
+
+Run the command below to crack James' password.
+```bash
+$ john /etc/shadow
+```
```
Comment (Ericgacoki) [inline content/articles/password-cracking-with-john-the-ripper/index.md:160]: These commands require `sudo` or `root` user permissions to work. Consider using the `sudo` prefix in the highlighted commands and add more information on the `passwd` command to make it clear.

I've also noticed that some images are not used. Please delete them.

**Review by Ericgacoki — APPROVED** [review]
Good work @Koros-Meshack.

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by Koros-Meshack** [comment]
Thanks for the Review @Ericgacoki

**Review by louisefindlay23 — CHANGES_REQUESTED** [review]
@Koros-Meshack, great tutorial. It's very easy to follow even for beginners but it would be nice if you went into extra detail about the background because some readers may not know much about password cracking, hashes and dictionary attacks.

**Suggested Topic Areas:** In-depth background about password cracking, what is a password hash, what is a dictionary attack and how they are related. Perhaps a bit of background into what penetration testing is and how Kali Linux helps with this?

**Comment by Koros-Meshack** [comment]
All updates made @louisefindlay23 
Please proceed to the final review
cc - @Ericgacoki

**Review by louisefindlay23 — APPROVED** [review]
@Koros-Meshack, thanks for adding an introduction with extra information. This gives your article that bit of extra value and also helps readers who may be new to the cyber security field to understand the terminology used.

Thanks, @Ericgacoki for peer reviewing.

**Pull request merged**